core/rawdb/accessors_chain.go

```diff
@@ -657,33 +657,35 @@ func WriteBlock(db ethdb.KeyValueWriter, block *types.Block) {
 
 // WriteAncientBlock writes entire block data into ancient store and returns the total written size.
 func WriteAncientBlock(db ethdb.AncientWriter, block *types.Block, receipts types.Receipts, td *big.Int) int {
-	// Encode all block components to RLP format.
-	headerBlob, err := rlp.EncodeToBytes(block.Header())
-	if err != nil {
-		log.Crit("Failed to RLP encode block header", "err", err)
+	batch := db.NewAncientBatch()
+
+	num := block.NumberU64()
+	if err := batch.AppendRaw(freezerHashTable, num, block.Hash().Bytes()); err != nil {
+		panic("Failed to RLP encode block hash: " + err.Error())
 	}
-	bodyBlob, err := rlp.EncodeToBytes(block.Body())
-	if err != nil {
-		log.Crit("Failed to RLP encode body", "err", err)
+	if err := batch.Append(freezerHeaderTable, num, block.Header()); err != nil {
+		panic("Failed to RLP encode block header: " + err.Error())
```

Comment:
Technically, we don't _know_ it was the RLP that failed, so all of these panics should say e.g. "Failed to batch-append XXX"